docs/testrunner-toolkit/configuration/puppeteer.md

```diff
@@ -1,73 +1,552 @@
+      - junit.xml
+    directory: ./artifacts/
+```
+---
+
+### `download`
+<p><small>| OPTIONAL | OBJECT |</small></p>
+
+Specifies the settings related to downloading artifacts from tests run by `saucectl`.
+
+```yaml
+  download:
+    when: always
+    match:
+      - junit.xml
+    directory: ./artifacts/
+```
+---
+
+#### `when`
+<p><small>| OPTIONAL | STRING |</small></p>
+
+Specifies when and under what circumstances to download artifacts. Valid values are:
+
+* `always`: Always download artifacts.
+* `never`: Never download artifacts.
+* `pass`: Download artifacts for passing suites only.
+* `fail`: Download artifacts for failed suites only.
+
+```yaml
+    when: always
+```
+---
+
+#### `match`
+<p><small>| OPTIONAL | STRING/ARRAY |</small></p>
 
-__Type__: *object*
+Specifies which artifacts to download based on whether they match the name or file type pattern provided. Supports the wildcard character `*`.
+
+```yaml
+  match:
+    - junit.xml
+    - *.log
+```
+---
+
+#### `directory`
+<p><small>| OPTIONAL | STRING |</small></p>
+
+Specifies the path to the folder location in which to download artifacts. A separate subdirectory is generated in this location for each suite for which artifacts are downloaded.
+
+```yaml
+    directory: ./artifacts/
+```
+---
+
+## `puppeteer`
+<p><small>| REQUIRED | OBJECT |</small></p>
+
+The parent property containing the details specific to the Puppeteer project.
 
-__Example__:
 ```yaml
 puppeteer:
-  version: ##VERSION##
+  version: 9.1.1
 ```
+---
 
 ### `version`
+<p><small>| REQUIRED | STRING |</small></p>
 
-__Description__: Version of `puppeteer` to use during tests.
-
-__Type__: *string*
+The version of Puppeteer that is compatible with the tests defined in this file.
 
-__Example__:
 ```yaml
-  version: ##VERSION##
+  version: 9.1.1
 ```
+---
 
 ## `suites`
+<p><small>| REQUIRED | OBJECT |</small></p>
+
+The set of properties providing details about the test suites to run. May contain multiple suite definitions. See the full [example config](#example-configuration) for an illustration of multiple suite definitions.
+
+---
 
-__Description__: Field for defining test suite details such as the suite `name` and desired `browser`.
 
-__Type__: *object*
+### `name`
+<p><small>| REQUIRED | STRING |</small></p>
+
+The name of the test suite, which will be reflected in the results and related artifacts.
+
+```yaml
+  - name: "saucy test"
+```
+---
+
+### `env`
+<p><small>| OPTIONAL | OBJECT |</small></p>
+
+A property containing one or more environment variables that may be referenced in the tests for this suite. Expanded environment variables are supported.
+
+```yaml
+  env:
+    hello: world
+    my_var: $MY_VAR
+```
+---
+
+### `browser`
+<p><small>| OPTIONAL | STRING |</small></p>
+
+The name of the browser in which to run this test suite.
 
-__Example__:
 ```yaml
-suites:
-  - name: "chrome"
-    testMatch: ["**/*.test.js"]
     browser: "chrome"
 ```
+---
 
-### `name`
+### `browserVersion`
```

Comment:
This option is not supported on puppeteer (as it's available only on cloud, and there is currently no support of puppeteer in cloud)